pkg/remote/remote.go

```diff
@@ -186,7 +187,11 @@ func (r *Remote) Fetch(goGetterSrc string, cacheDirOpt ...string) (string, error
 	replacer := strings.NewReplacer(":", "", "//", "_", "/", "_", ".", "_")
 	dirKey := replacer.Replace(srcDir)
 	if len(query) > 0 {
-		paramsKey := strings.Replace(query, "&", "_", -1)
+		q, _ := neturl.ParseQuery(query)
```

Comment:
According to https://github.com/hashicorp/go-getter/blob/4553965d9c4a8d99bd0d381c1180c08e07eff5fd/README.md#protocol-specific-options-1 there are multiple supported protocols like `git`, `http`, `s3` and `gcs`. All this protocols can include some sensitive information in any URL part.
Instead of redacting ever possible query parameter the whole URL could be hashed with `sha256`. This will fix the two problems mentioned in this pull request.
1. Don't leak sensitive information
2. Avoid hitting the max length of a folder name